receiver/couchbasereceiver/factory.go

```diff
@@ -0,0 +1,56 @@
+// Copyright  The OpenTelemetry Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package couchbasereceiver // import "github.com/open-telemetry/opentelemetry-collector-contrib/receiver/couchbasereceiver"
+
+import (
+	"context"
+	"time"
+
+	"go.opentelemetry.io/collector/component"
+	"go.opentelemetry.io/collector/config"
+	"go.opentelemetry.io/collector/config/confighttp"
+	"go.opentelemetry.io/collector/config/configtls"
+	"go.opentelemetry.io/collector/consumer"
+	"go.opentelemetry.io/collector/receiver/receiverhelper"
+	"go.opentelemetry.io/collector/receiver/scraperhelper"
+)
+
+const typeStr = "couchbase"
+
+// NewFactory creates a new receiver factory
+func NewFactory() component.ReceiverFactory {
+	return receiverhelper.NewFactory(
+		typeStr,
+		createDefaultConfig,
+		receiverhelper.WithMetrics(createMetricsReceiver))
+}
+
+func createDefaultConfig() config.Receiver {
+	return &Config{
+		ScraperControllerSettings: scraperhelper.ScraperControllerSettings{
+			ReceiverSettings:   config.NewReceiverSettings(config.NewComponentID(typeStr)),
+			CollectionInterval: 10 * time.Second,
+		},
+		HTTPClientSettings: confighttp.HTTPClientSettings{
+			TLSSetting: configtls.TLSClientSetting{},
```

Comment:
For the next PR, could you confirm this is actually needed? Also, it might be a good idea to require TLS when username/password is specified. There's really no excuses today to not use TLS, especially when credentials are involved. We have something like this in the following extension as well:

https://github.com/open-telemetry/opentelemetry-collector-contrib/blob/6f803948461e7256a7228f001343c1f5dcfae40e/extension/bearertokenauthextension/bearertokenauth.go#L40-L43